be/src/exec/vectorized/hdfs_scanner_text.cpp

```diff
@@ -99,16 +99,28 @@ Status HdfsScannerCSVReader::_fill_buffer() {
 Status HdfsTextScanner::do_init(RuntimeState* runtime_state, const HdfsScannerParams& scanner_params) {
     TTextFileDesc text_file_desc = _scanner_params.scan_ranges[0]->text_file_desc;
     _field_delimiter = text_file_desc.field_delim;
-    // we should cast string to char now since csv reader only support record delimiter by char
+
+    // we should cast string to char now since csv reader only support record, collection, mapkey delimiter by char
     _record_delimiter = text_file_desc.line_delim.front();
+    _collection_delimiter = text_file_desc.collection_delim.front();
+    _mapkey_delimiter = text_file_desc.mapkey_delim.front();
+
+    // delimiters will not be empty
+    DCHECK(!_field_delimiter.empty());
+    DCHECK(!_record_delimiter.empty());
+    DCHECK(!_collection_delimiter.empty());
+    DCHECK(!_mapkey_delimiter.empty());
     return Status::OK();
 }
 
 Status HdfsTextScanner::do_open(RuntimeState* runtime_state) {
     RETURN_IF_ERROR(_create_or_reinit_reader());
     SCOPED_RAW_TIMER(&_stats.reader_init_ns);
-    for (int i = 0; i < _scanner_params.materialize_slots.size(); i++) {
-        auto slot = _scanner_params.materialize_slots[i];
+    for (const auto slot : _scanner_params.materialize_slots) {
+        // TODO: Why slot may be nullptr?
+        if (slot == nullptr) {
```

Comment:
You could use `DCHECK` here, slot should not be null

Reply:
The previous pr has already discussed it.
<img width="1012" alt="image" src="https://user-images.githubusercontent.com/18729228/180944790-97e00eab-1a41-4c1c-b085-04a3ed2eba1a.png">